backend/assessments/1/src/main/java/org/example/Main.java

```diff
@@ -0,0 +1,220 @@
+            logger.logInfo("");
+            logger.logInfo(nameString + p.getName());
+            logger.logInfo(matchString + p.getMatches());
+            logger.logInfo(runString + p.getRuns());
+            logger.logInfo("Average: " + p.getAverage());
+            logger.logInfo("Strike Rate: " + p.getStrikeRate());
+        }
+        List<Player> topWicketTakers = team.getTeamPlayers().stream().sorted((player, t1) -> (int) (t1.getWickets() - player.getWickets())).limit(1).toList();
+        logger.logInfo("");
+        logger.logInfo("Top wicket taker for " + teamName +": ");
+        for(Player p: topWicketTakers){
+            logger.logInfo("");
+            logger.logInfo(nameString + p.getName());
+            logger.logInfo(matchString + p.getMatches());
+            logger.logInfo(wicketString + p.getWickets());
+        }
+    }
+
+    public static void displayTop3(){
+        List<Player> topRunScorers = players.stream().sorted((player, t1) -> (int) (t1.getRuns() - player.getRuns())).limit(3).toList();
+
+
+        logger.logInfo("Top 3 run scorers of the season: ");
+        for(Player p: topRunScorers){
+            logger.logInfo("");
+            logger.logInfo(nameString + p.getName());
+            logger.logInfo(matchString + p.getMatches());
+            logger.logInfo(runString + p.getRuns());
+            logger.logInfo("Average: " + p.getAverage());
+            logger.logInfo("Strike Rate: " + p.getStrikeRate());
+        }
+        List<Player> topWicketTakers = players.stream().sorted((player, t1) -> (int) (t1.getWickets() - player.getWickets())).limit(3).toList();
+        logger.logInfo("");
+        logger.logInfo("Top 3 wicket takers of the season: ");
+        for(Player p: topWicketTakers){
+            logger.logInfo("");
+            logger.logInfo(nameString + p.getName());
+            logger.logInfo(matchString + p.getMatches());
+            logger.logInfo(wicketString + p.getWickets());
+        }
+    }
+
+    public static void generateSchedule(int matchCount){
+        if(matchCount == 1){
+            makeMatch(date,time2);
+        }
+        else {
+            makeMatch(date,time1);
+            makeMatch(date,time2);
+        }
+    }
+
+    public static void makeMatch(String date, String time){
+        logger.logInfo("Enter home team name: ");
+        String homeTeamName = scanner.nextLine();
+        logger.logInfo("Enter away team name: ");
+        String awayTeamName = scanner.nextLine();
+        Team homeTeam = teamFromName.get(homeTeamName);
+        Team awayTeam = teamFromName.get(awayTeamName);
+        if(homeTeam.getHomeMatches().contains(awayTeam)){
+            logger.logInfo("Cannot schedule this match");
+        }
+        else{
+            Match match = new Match(homeTeam,awayTeam,date,time);
+            matches.add(match);
+            homeTeam.getHomeMatches().add(awayTeam);
+            date = DateIncrementer.addOneDay(date);
+        }
+    }
```

Comment:
You were expected to generate schedules for all teams. Also jumble the matches so that same team don't have multiple matches together